src/core/stdc/stdio.d

```diff
@@ -1410,16 +1446,71 @@ else version (CRuntime_Microsoft)
     int  vsnprintf(scope char* s, size_t n, scope const char* format, va_list arg);
   }
 
+    //
+    // Microsoft under-the-hood C I/O functions. Uses _iobuf* for the unshared
+    // version of FILE*, usable when the FILE is locked.
+    //
+    import core.stdc.stddef : wchar_t;
+    import core.stdc.wchar_ : wint_t;
+
+    version (MinGW)
+    {
+        private int _filbuf(_iobuf*);
+        private int _flsbuf(int, _iobuf*);
+
+        extern (D)
+        {
+            ///
+            int _fputc_nolock()(int c, _iobuf* fp)
+            {
+                pragma(inline, true);
+                fp._cnt--;
+                if (fp._cnt >= 0)
+                {
+                    *fp._ptr++ = cast(char)c;
+                    return c & 0xFF;
+                }
+                else
+                    return _flsbuf(c, fp);
+            }
+            ///
+            int _fgetc_nolock()(_iobuf* fp)
+            {
+                pragma(inline, true);
+                fp._cnt--;
+                if (fp._cnt >= 0)
+                {
+                    immutable c = *fp._ptr++;
+                    return c & 0xFF;
+                }
+                else
+                    return _filbuf(fp);
+            }
```

Comment:
We actually already have fallbacks for these (were macros for MSVC 2013 and older, symbols since MSVC 2015/UCRT): https://github.com/dlang/druntime/blob/master/src/rt/msvc.d#L145-L169 - not sure if MinGW supports the embedded linker directives though.

Anyway, why make them a template, why not simply `extern(D) pragma(inline, true) int ...`?

Reply:
GCC doesn't support these linker directives.  I can get around it in another way though (knowing that gcc always links in libmsvcrt.a) and just leave druntime assuming that these always exist (even though the wide char variants are UCRT functions).

Reply:
I've implemented this function in an msvc support module in gdc.